include/CL/cl2.hpp

```diff
@@ -6729,6 +6754,62 @@ class Program : public detail::Wrapper<cl_program>
         }
         return CL_SUCCESS;
     }
+
+#if CL_HPP_TARGET_OPENCL_VERSION >= 220
+    /*! \brief Registers a callback function to be called when destructors for
+     *         program scope global variables are complete and before the
+     *         program is released.
+     *
+     *  Wraps clSetProgramReleaseCallback().
+     *
+     *  Each call to this function registers the specified user callback function
+     *  on a callback stack associated with program. The registered user callback
+     *  functions are called in the reverse order in which they were registered.
+     */
+    cl_int setReleaseCallback(
+        void (CL_CALLBACK * pfn_notify)(cl_program program, void * user_data),
+        void * user_data = NULL)
+    {
+        return detail::errHandler(
+            ::clSetProgramReleaseCallback(
+                object_,
+                pfn_notify,
+                user_data),
+            __SET_PROGRAM_RELEASE_CALLBACK_ERR);
+    }
+
+    /*! \brief Sets a SPIR-V specialization constant.
+     *
+     *  Wraps clSetProgramSpecializationConstant().
+     */
+    template <typename T>
+    typename std::enable_if<!std::is_pointer<T>::value, cl_int>::type
```

Comment:
Why did you disable it for pointers?

Reply:
The SPIR-V spec for [OpSpecConstant](https://www.khronos.org/registry/spir-v/specs/unified1/SPIRV.html#OpSpecConstant) says:

> Result Type must be a scalar integer type or floating-point type.

I take this to mean that a spec constant cannot be a pointer, so pointers should be disabled.  Does that sound correct?

Reply:
Given that the OpenCL entrypoint just allows raw data to be copied (and doesn't really care about type), one might argue that declaring an integer on the SPIR-V side and storing, say an SVM pointer that could be cast back to a pointer after, is technically valid. That being said, people doing weird things like this can still use the other overload and I think I prefer what you've done.